pkg/pillar/types/errortime.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"reflect"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Common error with timestamp
+
+// ErrorAndTime is used by many EVE agents
+type ErrorAndTime struct {
+	Error     string
```

Comment:
Why not just use error? Why keep it as a string? Most of the cases, we are keeping the error directly - right?

Reply:
It would be a lot more change to track this as an error type. All of the callers are used to passing in a string. And it doesn't buy us anything to make it an error type AFAICT.